test/src/unit-vfs.cc

```diff
@@ -202,6 +202,81 @@ TEST_CASE("VFS: Test long local paths", "[vfs][long-paths]") {
   }
 }
 
+TEST_CASE("VFS: copy_file", "[vfs][copy_file]") {
+  LocalFsTest src_fs({0}), dst_fs({0});
+  URI src_path = src_fs.temp_dir_.add_trailing_slash();
+  URI dst_path = dst_fs.temp_dir_.add_trailing_slash();
+
+  ThreadPool compute_tp(4);
+  ThreadPool io_tp(4);
+  Config config = set_config_params();
+  VFS vfs{
+      &g_helper_stats, g_helper_logger().get(), &compute_tp, &io_tp, config};
+
+  size_t test_str_size = 0;
+  SECTION("Filesize = 0 MB") {
+    test_str_size = 0;
+  }
+  SECTION("Filesize = 1 MB") {
+    test_str_size = 1048576;
+  }
+  SECTION("Filesize = 10 MB") {
+    test_str_size = 10 * 1048576;
+  }
+  SECTION("Filesize = 100 MB") {
+    test_str_size = 100 * 1048576;
+  }
+  SECTION("Filesize = 150 MB") {
+    test_str_size = 150 * 1048576;
+  }
+  std::string test_chars = "abcdefghij";
```

Comment:
This is an improvement, but... the buffer sizes are a multiple of 10, aren't they?  This will have the same issue as before where all the buffers look the same so out-of-order reads or duplicate buffers will still pass the test.

I really do want to see a non-repeating sequence here. Random data is one way but not the only way.  Something like `"a", "bb", "ccc", "dddd", ..., "zzzzzzzzzzzzzzzzzzzzzzzzzz", "aa", "bbb", ...` would be fine as well.